### PR TITLE
Ensure that the platform view is created and destroyed when running the shell unittests.

### PR DESCRIPTION
**Pull request by chinmaygarde**
Also adds a failing test for https://github.com/flutter/flutter/issues/21398 that will be address in a subsequent patch.

### CODE REVIEW

File: shell/common/shell_unittests.cc [hunk 1]
```diff
@@ -14,6 +14,7 @@
 #include "flutter/shell/common/rasterizer.h"
 #include "flutter/shell/common/shell.h"
 #include "flutter/shell/common/thread_host.h"
+#include "flutter/shell/gpu/gpu_surface_software.h"
```
Comment (liyuqian) [inline shell/common/shell_unittests.cc]: The following is a general comment on naming that does not need to be addressed in this PR:

The name of "gpu_surface_software" seems a little strange: if it's software, it's not a gpu surface...

This issue exists in many places such as "gpu thread" which may or may not involve gpu. Maybe at some point, we should substitute all "gpu" with "render" so they become "render_surface_software" and "render thread".

File: shell/common/shell_unittests.cc [hunk 2]
```diff
@@ -23,6 +24,64 @@
...
+
+  // |GPUSurfaceSoftwareDelegate|
+  virtual bool PresentBackingStore(sk_sp<SkSurface> backing_store) override {
+    return true;
+  }
+
+  FML_DISALLOW_COPY_AND_ASSIGN(TestPlatformView);
+};
+
+bool ValidateShell(Shell* shell) {
```
Comment (liyuqian) [inline shell/common/shell_unittests.cc]: Nit: in Skia, we tend to put `static` before these functions (e.g., `static bool ValidateShell`) that are only used in this cc file.

Comment (chinmaygarde) [reply shell/common/shell_unittests.cc]: Done

File: shell/common/shell_unittests.cc (hunk 1 above)
Comment (chinmaygarde) [reply shell/common/shell_unittests.cc]: Agreed. The support for various backends using different client rendering APIs was added much later and these naming patterns are an unfortunate relic of the old way of doing things. These APIs are not public however and so should be straightforward to rename to make more idiomatic.

File: shell/common/shell_unittests.cc (hunk 2 above)
Comment (chinmaygarde) [reply shell/common/shell_unittests.cc]: This is a pattern we follow in the engine too. I was just forgetful :(

**Pull request merged**